app/services/ticker.py

```diff
@@ -0,0 +1,183 @@
+import asyncio
+from datetime import datetime
+from decimal import getcontext, InvalidOperation, DivisionByZero
+import logging
+from queue import Queue, Empty as QueueEmpty
+from web3 import Web3
+
+from ..app import App
+from ..src.erc20_token import ERC20Token
+
+logger = logging.getLogger('services.ticker')
+logger.setLevel(logging.DEBUG)
+
+getcontext().prec = 8
+
+# TODO: finally extract these into constants
+ZERO_ADDR = "0x0000000000000000000000000000000000000000"
+ZERO_ADDR_BYTES = Web3.toBytes(hexstr=ZERO_ADDR)
+
+tokens_queue = Queue()
+# TODO: Populate from our own DB
+with open("tokens.json") as f:
+    import json
+    for token in json.load(f):
+        tokens_queue.put(token["addr"].lower())
+
+async def get_trades_volume(token_hexstr):
+    """
+    Given a token address, return the volume of trades for that token in the past
+    24 hrs from the instant of calling this function.
+
+    Returns a Record object with two columns: quote_volume and base_volume.
+    """
+
+    async with App().db.acquire_connection() as conn:
+        return await conn.fetchrow("""
+            SELECT
+                COALESCE(SUM(CASE WHEN "token_get" = $1 THEN "amount_give" ELSE "amount_get" END), 0) AS quote_volume,
+                COALESCE(SUM(CASE WHEN "token_get" = $1 THEN "amount_get" ELSE "amount_give" END), 0) AS base_volume
+            FROM trades
+            WHERE ("token_give" = $2 OR "token_get" = $2)
+                AND "date" >= NOW() - '1 day'::INTERVAL
+            """,
+            Web3.toBytes(hexstr=ZERO_ADDR), # TODO: We can probably do with just token address
```

Comment:
This is future proof allowing a secondary address to check against though.

Reply:
Fair enough!